ts/components/conversation/message/reactions/ReactionPopup.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { ReactElement, useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { getMessageById } from '../../../../data/data';
+import { PubKey } from '../../../../session/types/PubKey';
+import { nativeEmojiData } from '../../../../util/emoji';
+import { readableList } from '../../../../util/readableList';
+
+export type TipPosition = 'center' | 'left' | 'right';
+
+export const StyledPopupContainer = styled.div<{ tooltipPosition: TipPosition }>`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  width: 216px;
+  height: 72px;
+  z-index: 5;
+
+  background-color: var(--color-received-message-background);
+  color: var(--color-pill-divider-text);
+  box-shadow: 0px 0px 13px rgba(0, 0, 0, 0.51);
+  font-size: 12px;
+  font-weight: 600;
+  overflow-wrap: break-word;
+  padding: 16px;
+  border-radius: 12px;
+  cursor: pointer;
+
+  &:after {
+    content: '';
+    position: absolute;
+    top: calc(100% - 19px);
+    left: ${props => {
+      switch (props.tooltipPosition) {
+        case 'left':
+          return '24px';
+        case 'right':
+          return 'calc(100% - 48px)';
+        case 'center':
+        default:
+          return 'calc(100% - 100px)';
+      }
+    }};
+    width: 22px;
+    height: 22px;
+    background-color: var(--color-received-message-background);
+    transform: rotate(45deg);
+    border-radius: 3px;
+    transform: scaleY(1.4) rotate(45deg);
+    clip-path: polygon(100% 100%, 7.2px 100%, 100% 7.2px);
+  }
+`;
+
+const StyledEmoji = styled.span`
+  font-size: 36px;
+  margin-left: 8px;
+`;
+
+const renderContacts = (_contacts: string) => {
+  if (!_contacts) {
+    return <></>;
```

Comment:
From what I have read just doing an empty return or null would be better because React wouldn't have to process it. But if I had added the return type of `ReactElement` to this (like a regular component) you would have to return an empty fragment because otherwise you get a syntax error.

I have built up a habit of returning empty fragments because of this. But I have changed it to just a return in this case.